.github/workflows/create-release.yml

```diff
@@ -1,279 +1,539 @@
+          printf "ubi_images_need_update=%s\n" "${json_ubi_images_need_update}" >> "$GITHUB_OUTPUT"
+
+          json_ubi_image_previous_hash_codes=$(jq --compact-output --null-input '$ARGS.positional' --args -- "${ubi_image_previous_hash_codes[@]}")
+          printf "ubi_image_previous_hash_codes=%s\n" "${json_ubi_image_previous_hash_codes}" >> "$GITHUB_OUTPUT"
 
   create_stack:
     name: Create Stack
+    needs: poll_ubi_images
+    if: ${{ ! ( needs.poll_ubi_images.outputs.ubi_images_need_update == '[]' && github.event_name == 'schedule' ) }}
+
     runs-on: ubuntu-22.04
     steps:
-    - name: Checkout
-      uses: actions/checkout@v3
-
-    - name: Create stack
-      id: create-stack
-      uses: paketo-buildpacks/github-config/actions/stack/create-stack@main
-
-    - name: Upload build image
-      uses: actions/upload-artifact@v3
-      with:
-        name: current-build-image
-        path: build/build.oci
-
-    - name: Upload base run image
-      uses: actions/upload-artifact@v3
-      with:
-        name: current-run-image
-        path: build/run.oci
-
-    - name: Upload nodejs-16 run image
-      uses: actions/upload-artifact@v3
-      with:
-        name: current-nodejs-16-run-image
-        path: build-nodejs-16/run.oci
-
-    - name: Upload nodejs-18 run image
-      uses: actions/upload-artifact@v3
-      with:
-        name: current-nodejs-18-run-image
-        path: build-nodejs-18/run.oci
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Create stack
+        id: create-stack
+        run: |
+          ./scripts/create.sh
+
+      ## build image
+      - name: Upload build image
+        uses: actions/upload-artifact@v3
+        with:
+          name: current-build-image
+          path: ./build/build.oci
+
+      - name: Get build image hash code
+        run: |
+          mkdir -p hash-codes/build
+
+          ubi_images_need_update=($(echo '${{ needs.poll_ubi_images.outputs.ubi_images_need_update }}' | jq -r '.[]'))
+          ubi_image_previous_hash_codes=($(echo '${{ needs.poll_ubi_images.outputs.ubi_image_previous_hash_codes }}' | jq -r '.[]'))
+
+          current_build_sha256_image_hash_code=""
+          if echo "${ubi_images_need_update[@]}" | grep -qw "node-build"; then
+            docker pull registry.access.redhat.com/ubi8/ubi-minimal
+            current_build_sha256_image_hash_code=$(docker image inspect registry.access.redhat.com/ubi8/ubi-minimal --format '{{json .RepoDigests}}' | jq -r '.[0]')
+          else
+            for ((i = 0; i < ${#ubi_image_previous_hash_codes[@]}; i += 2)); do
+              if [ "${ubi_image_previous_hash_codes[i]}" == "node-build" ]; then
+                current_build_sha256_image_hash_code=${ubi_image_previous_hash_codes[i + 1]}
+              fi
+            done
+          fi
+
+          echo $current_build_sha256_image_hash_code > hash-codes/build/build.oci.sha256
```

Comment:
it might be more ideal to turn this into an action for generating the hashcode since its kind of complicated and duplicated for every image (4x)

Reply:
Better yet actually, can we reuse the hash codes we got from the `Compare previous and current sha256 hash codes` step? Rather than redoing the docker pull and everything here

Reply:
I avoided creating it in a github action as is only specific to this repo and not that much code, so it might turn out more overhead in terms of maintenance on a piece of code that will only change once a year, I'm 100% on the part of code reusability. 
Also I see a bit similar pattern on https://github.com/paketo-buildpacks/jammy-base-stack/blob/5735161dd48e081b23647792cbf2e58191d23a0d/.github/workflows/create-release.yml#L110-L123C11 with the only difference there is no for loop and is not repeated across steps.

This piece of code is a bit complicated due to we are trying to avoid docker pull by fetching the hash code from the array  `ubi_image_previous_hash_codes` generated from previous step, but after the improvement you suggested (using skopeo instead of docker pull), that might not be necessary as the overhead of skopeo is very little compared to docker pull and that way we can replace the whole code of this step with this command `skopeo inspect --format "{{.Digest}}" docker://registry.access.redhat.com/ubi8/ubi-minimal > hash-codes/build/build.oci.sha256` 
We can go with a github action on the existing code, but on the other hand by using a github action we still need to do a request to fetch the github action which seems equivalent to the skopeo command in terms of performance with skopeo command being much less code to maintain.

Reply:
Alright, I think I'm with you on not converting this into an action, it's probably not worth the overhead. I am still wondering why we can't reuse the checksums from the `Compare previous and current sha256 hash codes` step by storing them in an output and then referencing them in this job (rather than re-doing the skopeo call)? Sorry if I missed this in your previous message

Reply:
I see your point, good catch! I updated the PR with this commit. https://github.com/paketo-community/ubi-base-stack/pull/7/commits/14995fddd87b0ab234f378f85a6d425ac3ee213c
So I'm storing the hash codes on the `ubi_image_current_hash_codes` array from previous steps, specifically on the `Compare previous and current sha256 hash codes` (as you said) and on the `get * image hash code` steps I fetch them from the `ubi_image_current_hash_codes` array and upload them. 
Also, I ended up removing the `ubi_image_previous_hash_codes` array as is no longer necessary after your suggestion.